apps/functions-subscription/src/use-cases/process-activation-job.ts

```diff
@@ -12,12 +12,16 @@ export const processActivationJob = (
 ) =>
   pipe(
     RTE.ask<Env>(),
-    RTE.flatMapTaskEither(({ activationRequestRepository }) =>
-      pipe(
-        activationRequestRepository.list(job.trialId, maxFetchSize),
+    RTE.flatMapTaskEither(({ activationRequestRepository }) => {
+      const itemsToFetch = Math.min(
+        maxFetchSize,
+        job.usersToActivate - job.usersActivated,
+      );
+      return pipe(
+        activationRequestRepository.list(job.trialId, itemsToFetch),
         TE.flatMap((activationRequests) =>
           activationRequestRepository.activate(job, activationRequests),
         ),
-      ),
-    ),
+      );
+    }),
```

Comment:
**nitpick**: You can use `RTE.let`. Feel free to keep the original version :)

```TypeScript
  pipe(
    RTE.ask<Env>(),
    RTE.let('limit', () =>
      Math.min(maxFetchSize, job.usersToActivate - job.usersActivated),
    ),
    RTE.flatMapTaskEither(({ activationRequestRepository, limit }) =>
      pipe(
        activationRequestRepository.list(job.trialId, limit),
        TE.flatMap((activationRequests) =>
          activationRequestRepository.activate(job, activationRequests),
        ),
      ),
    ),
  );
```

Reply:
I like it, thanks for sharing!
I'm gonna change it in the next commit.

Reply:
✅